Description: Removed references to symbols not in official fribidi API
  This package uses symbols that weren't supposed to be exposed in fribidi.
  Upstream of fribidi at the time 0.10.9 was released did not know about hiding
  symbols. These symbols are now removed from fribidi 0.19.2. Please refer to:
  http://lists.freedesktop.org/archives/fribidi/2009-March/000565.html ,
  http://lists.alioth.debian.org/pipermail/debian-hebrew-package/2009-March/002330.html
Author: أحمد المحمودي (Ahmed El-Mahmoudy) <aelmahmoudy@users.sourceforge.net>
--- a/lib/Text/Bidi.pm
+++ b/lib/Text/Bidi.pm
@@ -528,7 +528,6 @@
 =cut
 
 sub caprtl_to_unicode {
-    Text::Bidi::private::fribidi_char_set_enter_cap_rtl();
     decode('UTF-32LE', Text::Bidi::private::fribidi_cap_rtl_to_unicode(@_))
 }
 
@@ -539,7 +538,6 @@
 =cut
 
 sub unicode_to_caprtl {
-    Text::Bidi::private::fribidi_char_set_enter_cap_rtl();
     Text::Bidi::private::fribidi_unicode_to_cap_rtl(encode('UTF-32LE', shift));
 }
 
--- a/lib/Text/Bidi/private.pm
+++ b/lib/Text/Bidi/private.pm
@@ -61,7 +61,6 @@
 *fribidi_get_mirror_char = *Text::Bidi::privatec::fribidi_get_mirror_char;
 *fribidi_set_mirroring = *Text::Bidi::privatec::fribidi_set_mirroring;
 *fribidi_set_reorder_nsm = *Text::Bidi::privatec::fribidi_set_reorder_nsm;
-*fribidi_char_set_enter_cap_rtl = *Text::Bidi::privatec::fribidi_char_set_enter_cap_rtl;
 *fribidi_cap_rtl_to_unicode = *Text::Bidi::privatec::fribidi_cap_rtl_to_unicode;
 *fribidi_unicode_to_cap_rtl = *Text::Bidi::privatec::fribidi_unicode_to_cap_rtl;
 
@@ -218,5 +217,4 @@
 *FRIBIDI_TYPE_SOT = *Text::Bidi::privatec::FRIBIDI_TYPE_SOT;
 *FRIBIDI_TYPE_EOT = *Text::Bidi::privatec::FRIBIDI_TYPE_EOT;
 *FRIBIDI_TYPES_COUNT = *Text::Bidi::privatec::FRIBIDI_TYPES_COUNT;
-*fribidi_prop_to_type = *Text::Bidi::privatec::fribidi_prop_to_type;
 1;
--- a/private.c
+++ b/private.c
@@ -1407,18 +1407,16 @@
 /* -------- TYPES TABLE (BEGIN) -------- */
 
 #define SWIGTYPE_p_FriBidiCharSet swig_types[0]
-#define SWIGTYPE_p_FriBidiRunType swig_types[1]
-#define SWIGTYPE_p__FriBidiList swig_types[2]
-#define SWIGTYPE_p_char swig_types[3]
-#define SWIGTYPE_p_int swig_types[4]
-#define SWIGTYPE_p_short swig_types[5]
-#define SWIGTYPE_p_signed_char swig_types[6]
-#define SWIGTYPE_p_unsigned_char swig_types[7]
-#define SWIGTYPE_p_unsigned_int swig_types[8]
-#define SWIGTYPE_p_unsigned_short swig_types[9]
-#define SWIGTYPE_p_void swig_types[10]
-static swig_type_info *swig_types[12];
-static swig_module_info swig_module = {swig_types, 11, 0, 0, 0, 0};
+#define SWIGTYPE_p_char swig_types[1]
+#define SWIGTYPE_p_int swig_types[2]
+#define SWIGTYPE_p_short swig_types[3]
+#define SWIGTYPE_p_signed_char swig_types[4]
+#define SWIGTYPE_p_unsigned_char swig_types[5]
+#define SWIGTYPE_p_unsigned_int swig_types[6]
+#define SWIGTYPE_p_unsigned_short swig_types[7]
+#define SWIGTYPE_p_void swig_types[8]
+static swig_type_info *swig_types[9];
+static swig_module_info swig_module = {swig_types, 9, 0, 0, 0, 0};
 #define SWIG_TypeQuery(name) SWIG_TypeQueryModule(&swig_module, &swig_module, name)
 #define SWIG_MangledTypeQuery(name) SWIG_MangledTypeQueryModule(&swig_module, &swig_module, name)
 
@@ -1455,7 +1453,7 @@
 #define Newx(A,B,C) New(42,A,B,C)
 #endif
 #include <fribidi/fribidi.h>
-FRIBIDI_API void log2vis ( /* input */
+FRIBIDI_ENTRY void log2vis ( /* input */
                            FriBidiChar *str,
                            FriBidiStrIndex len,
                            /* output */
@@ -1772,11 +1770,6 @@
     croak("Value is read-only.");
     return 0;
 }
-SWIGCLASS_STATIC int _wrap_fribidi_prop_to_type_get(pTHX_ SV *sv, MAGIC *SWIGUNUSEDPARM(mg)) {
-  MAGIC_PPERL
-  sv_setiv(SvRV(sv),PTR2IV(fribidi_prop_to_type));
-  return 1;
-}
 
 
 
@@ -1979,548 +1972,6 @@
 }
 
 
-XS(_wrap_fribidi_wcwidth) {
-  {
-    FriBidiChar arg1 ;
-    int result;
-    unsigned int val1 ;
-    int ecode1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: fribidi_wcwidth(ch);");
-    }
-    ecode1 = SWIG_AsVal_unsigned_SS_int SWIG_PERL_CALL_ARGS_2(ST(0), &val1);
-    if (!SWIG_IsOK(ecode1)) {
-      SWIG_exception_fail(SWIG_ArgError(ecode1), "in method '" "fribidi_wcwidth" "', argument " "1"" of type '" "FriBidiChar""'");
-    } 
-    arg1 = (FriBidiChar)(val1);
-    result = (int)fribidi_wcwidth(arg1);
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_fribidi_wcswidth) {
-  {
-    FriBidiChar *arg1 = (FriBidiChar *) 0 ;
-    FriBidiStrIndex arg2 ;
-    int result;
-    char *buf1 = 0 ;
-    size_t size1 = 0 ;
-    FriBidiStrIndex *_global_p_len ;
-    int argvi = 0;
-    dXSARGS;
-    
-    
-    arg2 = 0;
-    _global_p_len = &arg2;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: fribidi_wcswidth(str);");
-    }
-    buf1 = SvPV(ST(0), size1);
-    *_global_p_len = (FriBidiStrIndex)(size1/sizeof(FriBidiChar));
-    arg1 = (FriBidiChar *)(buf1);
-    result = (int)fribidi_wcswidth((unsigned int const *)arg1,arg2);
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_fribidi_wcswidth_cjk) {
-  {
-    FriBidiChar *arg1 = (FriBidiChar *) 0 ;
-    FriBidiStrIndex arg2 ;
-    int result;
-    char *buf1 = 0 ;
-    size_t size1 = 0 ;
-    FriBidiStrIndex *_global_p_len ;
-    int argvi = 0;
-    dXSARGS;
-    
-    
-    arg2 = 0;
-    _global_p_len = &arg2;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: fribidi_wcswidth_cjk(str);");
-    }
-    buf1 = SvPV(ST(0), size1);
-    *_global_p_len = (FriBidiStrIndex)(size1/sizeof(FriBidiChar));
-    arg1 = (FriBidiChar *)(buf1);
-    result = (int)fribidi_wcswidth_cjk((unsigned int const *)arg1,arg2);
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_fribidi_type_name) {
-  {
-    FriBidiCharType arg1 ;
-    char *result = 0 ;
-    int val1 ;
-    int ecode1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: fribidi_type_name(c);");
-    }
-    ecode1 = SWIG_AsVal_int SWIG_PERL_CALL_ARGS_2(ST(0), &val1);
-    if (!SWIG_IsOK(ecode1)) {
-      SWIG_exception_fail(SWIG_ArgError(ecode1), "in method '" "fribidi_type_name" "', argument " "1"" of type '" "FriBidiCharType""'");
-    } 
-    arg1 = (FriBidiCharType)(val1);
-    result = (char *)fribidi_type_name(arg1);
-    ST(argvi) = SWIG_FromCharPtr((const char *)result); argvi++ ;
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_FriBidiRunType_length_set) {
-  {
-    FriBidiRunType *arg1 = (FriBidiRunType *) 0 ;
-    FriBidiStrIndex arg2 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int val2 ;
-    int ecode2 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 2) || (items > 2)) {
-      SWIG_croak("Usage: FriBidiRunType_length_set(self,length);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p_FriBidiRunType, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "FriBidiRunType_length_set" "', argument " "1"" of type '" "FriBidiRunType *""'"); 
-    }
-    arg1 = (FriBidiRunType *)(argp1);
-    ecode2 = SWIG_AsVal_int SWIG_PERL_CALL_ARGS_2(ST(1), &val2);
-    if (!SWIG_IsOK(ecode2)) {
-      SWIG_exception_fail(SWIG_ArgError(ecode2), "in method '" "FriBidiRunType_length_set" "', argument " "2"" of type '" "FriBidiStrIndex""'");
-    } 
-    arg2 = (FriBidiStrIndex)(val2);
-    if (arg1) (arg1)->length = arg2;
-    
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_FriBidiRunType_length_get) {
-  {
-    FriBidiRunType *arg1 = (FriBidiRunType *) 0 ;
-    FriBidiStrIndex result;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: FriBidiRunType_length_get(self);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p_FriBidiRunType, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "FriBidiRunType_length_get" "', argument " "1"" of type '" "FriBidiRunType *""'"); 
-    }
-    arg1 = (FriBidiRunType *)(argp1);
-    result = (FriBidiStrIndex) ((arg1)->length);
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_FriBidiRunType_attribute_set) {
-  {
-    FriBidiRunType *arg1 = (FriBidiRunType *) 0 ;
-    void *arg2 = (void *) 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int res2 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 2) || (items > 2)) {
-      SWIG_croak("Usage: FriBidiRunType_attribute_set(self,attribute);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p_FriBidiRunType, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "FriBidiRunType_attribute_set" "', argument " "1"" of type '" "FriBidiRunType *""'"); 
-    }
-    arg1 = (FriBidiRunType *)(argp1);
-    res2 = SWIG_ConvertPtr(ST(1),SWIG_as_voidptrptr(&arg2), 0, SWIG_POINTER_DISOWN);
-    if (!SWIG_IsOK(res2)) {
-      SWIG_exception_fail(SWIG_ArgError(res2), "in method '" "FriBidiRunType_attribute_set" "', argument " "2"" of type '" "void *""'"); 
-    }
-    if (arg1) (arg1)->attribute = arg2;
-    
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_FriBidiRunType_attribute_get) {
-  {
-    FriBidiRunType *arg1 = (FriBidiRunType *) 0 ;
-    void *result = 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: FriBidiRunType_attribute_get(self);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p_FriBidiRunType, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "FriBidiRunType_attribute_get" "', argument " "1"" of type '" "FriBidiRunType *""'"); 
-    }
-    arg1 = (FriBidiRunType *)(argp1);
-    result = (void *) ((arg1)->attribute);
-    ST(argvi) = SWIG_NewPointerObj(SWIG_as_voidptr(result), SWIGTYPE_p_void, 0 | 0); argvi++ ;
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_new_FriBidiRunType) {
-  {
-    FriBidiRunType *result = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 0) || (items > 0)) {
-      SWIG_croak("Usage: new_FriBidiRunType();");
-    }
-    result = (FriBidiRunType *)(FriBidiRunType *) calloc(1, sizeof(FriBidiRunType));
-    ST(argvi) = SWIG_NewPointerObj(SWIG_as_voidptr(result), SWIGTYPE_p_FriBidiRunType, SWIG_OWNER | SWIG_SHADOW); argvi++ ;
-    XSRETURN(argvi);
-  fail:
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_delete_FriBidiRunType) {
-  {
-    FriBidiRunType *arg1 = (FriBidiRunType *) 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: delete_FriBidiRunType(self);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p_FriBidiRunType, SWIG_POINTER_DISOWN |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "delete_FriBidiRunType" "', argument " "1"" of type '" "FriBidiRunType *""'"); 
-    }
-    arg1 = (FriBidiRunType *)(argp1);
-    free((char *) arg1);
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap__FriBidiList_data_set) {
-  {
-    struct _FriBidiList *arg1 = (struct _FriBidiList *) 0 ;
-    void *arg2 = (void *) 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int res2 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 2) || (items > 2)) {
-      SWIG_croak("Usage: _FriBidiList_data_set(self,data);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p__FriBidiList, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "_FriBidiList_data_set" "', argument " "1"" of type '" "struct _FriBidiList *""'"); 
-    }
-    arg1 = (struct _FriBidiList *)(argp1);
-    res2 = SWIG_ConvertPtr(ST(1),SWIG_as_voidptrptr(&arg2), 0, SWIG_POINTER_DISOWN);
-    if (!SWIG_IsOK(res2)) {
-      SWIG_exception_fail(SWIG_ArgError(res2), "in method '" "_FriBidiList_data_set" "', argument " "2"" of type '" "void *""'"); 
-    }
-    if (arg1) (arg1)->data = arg2;
-    
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap__FriBidiList_data_get) {
-  {
-    struct _FriBidiList *arg1 = (struct _FriBidiList *) 0 ;
-    void *result = 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: _FriBidiList_data_get(self);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p__FriBidiList, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "_FriBidiList_data_get" "', argument " "1"" of type '" "struct _FriBidiList *""'"); 
-    }
-    arg1 = (struct _FriBidiList *)(argp1);
-    result = (void *) ((arg1)->data);
-    ST(argvi) = SWIG_NewPointerObj(SWIG_as_voidptr(result), SWIGTYPE_p_void, 0 | 0); argvi++ ;
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap__FriBidiList_next_set) {
-  {
-    struct _FriBidiList *arg1 = (struct _FriBidiList *) 0 ;
-    FriBidiList *arg2 = (FriBidiList *) 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    void *argp2 = 0 ;
-    int res2 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 2) || (items > 2)) {
-      SWIG_croak("Usage: _FriBidiList_next_set(self,next);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p__FriBidiList, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "_FriBidiList_next_set" "', argument " "1"" of type '" "struct _FriBidiList *""'"); 
-    }
-    arg1 = (struct _FriBidiList *)(argp1);
-    res2 = SWIG_ConvertPtr(ST(1), &argp2,SWIGTYPE_p__FriBidiList, SWIG_POINTER_DISOWN |  0 );
-    if (!SWIG_IsOK(res2)) {
-      SWIG_exception_fail(SWIG_ArgError(res2), "in method '" "_FriBidiList_next_set" "', argument " "2"" of type '" "FriBidiList *""'"); 
-    }
-    arg2 = (FriBidiList *)(argp2);
-    if (arg1) (arg1)->next = arg2;
-    
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap__FriBidiList_next_get) {
-  {
-    struct _FriBidiList *arg1 = (struct _FriBidiList *) 0 ;
-    FriBidiList *result = 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: _FriBidiList_next_get(self);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p__FriBidiList, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "_FriBidiList_next_get" "', argument " "1"" of type '" "struct _FriBidiList *""'"); 
-    }
-    arg1 = (struct _FriBidiList *)(argp1);
-    result = (FriBidiList *) ((arg1)->next);
-    ST(argvi) = SWIG_NewPointerObj(SWIG_as_voidptr(result), SWIGTYPE_p__FriBidiList, 0 | SWIG_SHADOW); argvi++ ;
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap__FriBidiList_prev_set) {
-  {
-    struct _FriBidiList *arg1 = (struct _FriBidiList *) 0 ;
-    FriBidiList *arg2 = (FriBidiList *) 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    void *argp2 = 0 ;
-    int res2 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 2) || (items > 2)) {
-      SWIG_croak("Usage: _FriBidiList_prev_set(self,prev);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p__FriBidiList, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "_FriBidiList_prev_set" "', argument " "1"" of type '" "struct _FriBidiList *""'"); 
-    }
-    arg1 = (struct _FriBidiList *)(argp1);
-    res2 = SWIG_ConvertPtr(ST(1), &argp2,SWIGTYPE_p__FriBidiList, SWIG_POINTER_DISOWN |  0 );
-    if (!SWIG_IsOK(res2)) {
-      SWIG_exception_fail(SWIG_ArgError(res2), "in method '" "_FriBidiList_prev_set" "', argument " "2"" of type '" "FriBidiList *""'"); 
-    }
-    arg2 = (FriBidiList *)(argp2);
-    if (arg1) (arg1)->prev = arg2;
-    
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap__FriBidiList_prev_get) {
-  {
-    struct _FriBidiList *arg1 = (struct _FriBidiList *) 0 ;
-    FriBidiList *result = 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: _FriBidiList_prev_get(self);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p__FriBidiList, 0 |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "_FriBidiList_prev_get" "', argument " "1"" of type '" "struct _FriBidiList *""'"); 
-    }
-    arg1 = (struct _FriBidiList *)(argp1);
-    result = (FriBidiList *) ((arg1)->prev);
-    ST(argvi) = SWIG_NewPointerObj(SWIG_as_voidptr(result), SWIGTYPE_p__FriBidiList, 0 | SWIG_SHADOW); argvi++ ;
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_new__FriBidiList) {
-  {
-    struct _FriBidiList *result = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 0) || (items > 0)) {
-      SWIG_croak("Usage: new__FriBidiList();");
-    }
-    result = (struct _FriBidiList *)(struct _FriBidiList *) calloc(1, sizeof(struct _FriBidiList));
-    ST(argvi) = SWIG_NewPointerObj(SWIG_as_voidptr(result), SWIGTYPE_p__FriBidiList, SWIG_OWNER | SWIG_SHADOW); argvi++ ;
-    XSRETURN(argvi);
-  fail:
-    SWIG_croak_null();
-  }
-}
-
-
-XS(_wrap_delete__FriBidiList) {
-  {
-    struct _FriBidiList *arg1 = (struct _FriBidiList *) 0 ;
-    void *argp1 = 0 ;
-    int res1 = 0 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: delete__FriBidiList(self);");
-    }
-    res1 = SWIG_ConvertPtr(ST(0), &argp1,SWIGTYPE_p__FriBidiList, SWIG_POINTER_DISOWN |  0 );
-    if (!SWIG_IsOK(res1)) {
-      SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "delete__FriBidiList" "', argument " "1"" of type '" "struct _FriBidiList *""'"); 
-    }
-    arg1 = (struct _FriBidiList *)(argp1);
-    free((char *) arg1);
-    
-    
-    
-    XSRETURN(argvi);
-  fail:
-    
-    SWIG_croak_null();
-  }
-}
-
-
 XS(_wrap_fribidi_log2vis_get_embedding_levels) {
   {
     FriBidiChar *arg1 = (FriBidiChar *) 0 ;
@@ -2752,72 +2203,6 @@
 }
 
 
-XS(_wrap_fribidi_get_types) {
-  {
-    FriBidiChar *arg1 = (FriBidiChar *) 0 ;
-    FriBidiStrIndex arg2 ;
-    FriBidiCharType *arg3 = (FriBidiCharType *) 0 ;
-    char *buf1 = 0 ;
-    size_t size1 = 0 ;
-    FriBidiStrIndex *_global_p_len ;
-    FriBidiCharType **temp3 ;
-    int argvi = 0;
-    dXSARGS;
-    
-    
-    arg2 = 0;
-    _global_p_len = &arg2;
-    
-    
-    temp3 = &arg3;
-    
-    if ((items < 1) || (items > 1)) {
-      SWIG_croak("Usage: fribidi_get_types(str);");
-    }
-    buf1 = SvPV(ST(0), size1);
-    *_global_p_len = (FriBidiStrIndex)(size1/sizeof(FriBidiChar));
-    arg1 = (FriBidiChar *)(buf1);
-    if ( 1 ) {
-      Newx(*temp3, ((*_global_p_len) + 1), FriBidiCharType);
-    }
-    fribidi_get_types(arg1,arg2,arg3);
-    
-    {
-      if ( 1 )
-      /*@SWIG:MXPUSHUA@*/
-      {
-        AV* tempav = newAV();
-        int i;
-        for(i=0 ; i < (*_global_p_len) ; i++)
-        av_push(tempav, newSVuv(arg3[i]));
-        /*@SWIG:MXPUSHR@*/
-        /*@SWIG:XPUSHS@*/
-        if (argvi >= items)
-        EXTEND(sp, 1);
-        ST(argvi) = sv_2mortal(newRV_noinc((SV *)tempav));
-        argvi++
-        /*@SWIG@*/
-        /*@SWIG@*/;
-      }
-      /*@SWIG@*/;
-    }
-    
-    
-    
-    if (arg3) Safefree(arg3);
-    
-    XSRETURN(argvi);
-  fail:
-    
-    
-    
-    if (arg3) Safefree(arg3);
-    
-    SWIG_croak_null();
-  }
-}
-
-
 XS(_wrap_fribidi_get_mirror_char) {
   {
     FriBidiChar arg1 ;
@@ -2912,24 +2297,6 @@
 }
 
 
-XS(_wrap_fribidi_char_set_enter_cap_rtl) {
-  {
-    fribidi_boolean result;
-    int argvi = 0;
-    dXSARGS;
-    
-    if ((items < 0) || (items > 0)) {
-      SWIG_croak("Usage: fribidi_char_set_enter_cap_rtl();");
-    }
-    result = (fribidi_boolean)fribidi_char_set_enter_cap_rtl();
-    
-    XSRETURN(argvi);
-  fail:
-    SWIG_croak_null();
-  }
-}
-
-
 XS(_wrap_fribidi_cap_rtl_to_unicode) {
   {
     char *arg1 = (char *) 0 ;
@@ -3044,8 +2411,6 @@
 /* -------- TYPE CONVERSION AND EQUIVALENCE RULES (BEGIN) -------- */
 
 static swig_type_info _swigt__p_FriBidiCharSet = {"_p_FriBidiCharSet", "enum FriBidiCharSet *|FriBidiCharSet *", 0, 0, (void*)0, 0};
-static swig_type_info _swigt__p_FriBidiRunType = {"_p_FriBidiRunType", "FriBidiRunType *", 0, 0, (void*)"Text::Bidi::private::FriBidiRunType", 0};
-static swig_type_info _swigt__p__FriBidiList = {"_p__FriBidiList", "_FriBidiList *", 0, 0, (void*)"Text::Bidi::private::_FriBidiList", 0};
 static swig_type_info _swigt__p_char = {"_p_char", "char *|FriBidiPropCharType *", 0, 0, (void*)0, 0};
 static swig_type_info _swigt__p_int = {"_p_int", "int *|FriBidiCharType *", 0, 0, (void*)0, 0};
 static swig_type_info _swigt__p_short = {"_p_short", "short *|fribidi_int16 *", 0, 0, (void*)0, 0};
@@ -3057,8 +2422,6 @@
 
 static swig_type_info *swig_type_initial[] = {
   &_swigt__p_FriBidiCharSet,
-  &_swigt__p_FriBidiRunType,
-  &_swigt__p__FriBidiList,
   &_swigt__p_char,
   &_swigt__p_int,
   &_swigt__p_short,
@@ -3070,8 +2433,6 @@
 };
 
 static swig_cast_info _swigc__p_FriBidiCharSet[] = {  {&_swigt__p_FriBidiCharSet, 0, 0, 0},{0, 0, 0, 0}};
-static swig_cast_info _swigc__p_FriBidiRunType[] = {  {&_swigt__p_FriBidiRunType, 0, 0, 0},{0, 0, 0, 0}};
-static swig_cast_info _swigc__p__FriBidiList[] = {  {&_swigt__p__FriBidiList, 0, 0, 0},{0, 0, 0, 0}};
 static swig_cast_info _swigc__p_char[] = {  {&_swigt__p_char, 0, 0, 0},{0, 0, 0, 0}};
 static swig_cast_info _swigc__p_int[] = {  {&_swigt__p_int, 0, 0, 0},{0, 0, 0, 0}};
 static swig_cast_info _swigc__p_short[] = {  {&_swigt__p_short, 0, 0, 0},{0, 0, 0, 0}};
@@ -3083,8 +2444,6 @@
 
 static swig_cast_info *swig_cast_initial[] = {
   _swigc__p_FriBidiCharSet,
-  _swigc__p_FriBidiRunType,
-  _swigc__p__FriBidiList,
   _swigc__p_char,
   _swigc__p_int,
   _swigc__p_short,
@@ -3105,37 +2464,16 @@
 }
 #endif
 static swig_variable_info swig_variables[] = {
-    { "Text::Bidi::privatec::fribidi_prop_to_type", MAGIC_CLASS swig_magic_readonly, MAGIC_CLASS _wrap_fribidi_prop_to_type_get,&SWIGTYPE_p_int },
 {0,0,0,0}
 };
 static swig_command_info swig_commands[] = {
 {"Text::Bidi::privatec::log2vis", _wrap_log2vis},
-{"Text::Bidi::privatec::fribidi_wcwidth", _wrap_fribidi_wcwidth},
-{"Text::Bidi::privatec::fribidi_wcswidth", _wrap_fribidi_wcswidth},
-{"Text::Bidi::privatec::fribidi_wcswidth_cjk", _wrap_fribidi_wcswidth_cjk},
-{"Text::Bidi::privatec::fribidi_type_name", _wrap_fribidi_type_name},
-{"Text::Bidi::privatec::FriBidiRunType_length_set", _wrap_FriBidiRunType_length_set},
-{"Text::Bidi::privatec::FriBidiRunType_length_get", _wrap_FriBidiRunType_length_get},
-{"Text::Bidi::privatec::FriBidiRunType_attribute_set", _wrap_FriBidiRunType_attribute_set},
-{"Text::Bidi::privatec::FriBidiRunType_attribute_get", _wrap_FriBidiRunType_attribute_get},
-{"Text::Bidi::privatec::new_FriBidiRunType", _wrap_new_FriBidiRunType},
-{"Text::Bidi::privatec::delete_FriBidiRunType", _wrap_delete_FriBidiRunType},
-{"Text::Bidi::privatec::_FriBidiList_data_set", _wrap__FriBidiList_data_set},
-{"Text::Bidi::privatec::_FriBidiList_data_get", _wrap__FriBidiList_data_get},
-{"Text::Bidi::privatec::_FriBidiList_next_set", _wrap__FriBidiList_next_set},
-{"Text::Bidi::privatec::_FriBidiList_next_get", _wrap__FriBidiList_next_get},
-{"Text::Bidi::privatec::_FriBidiList_prev_set", _wrap__FriBidiList_prev_set},
-{"Text::Bidi::privatec::_FriBidiList_prev_get", _wrap__FriBidiList_prev_get},
-{"Text::Bidi::privatec::new__FriBidiList", _wrap_new__FriBidiList},
-{"Text::Bidi::privatec::delete__FriBidiList", _wrap_delete__FriBidiList},
 {"Text::Bidi::privatec::fribidi_log2vis_get_embedding_levels", _wrap_fribidi_log2vis_get_embedding_levels},
 {"Text::Bidi::privatec::fribidi_remove_bidi_marks", _wrap_fribidi_remove_bidi_marks},
 {"Text::Bidi::privatec::fribidi_get_type", _wrap_fribidi_get_type},
-{"Text::Bidi::privatec::fribidi_get_types", _wrap_fribidi_get_types},
 {"Text::Bidi::privatec::fribidi_get_mirror_char", _wrap_fribidi_get_mirror_char},
 {"Text::Bidi::privatec::fribidi_set_mirroring", _wrap_fribidi_set_mirroring},
 {"Text::Bidi::privatec::fribidi_set_reorder_nsm", _wrap_fribidi_set_reorder_nsm},
-{"Text::Bidi::privatec::fribidi_char_set_enter_cap_rtl", _wrap_fribidi_char_set_enter_cap_rtl},
 {"Text::Bidi::privatec::fribidi_cap_rtl_to_unicode", _wrap_fribidi_cap_rtl_to_unicode},
 {"Text::Bidi::privatec::fribidi_unicode_to_cap_rtl", _wrap_fribidi_unicode_to_cap_rtl},
 {0,0}
@@ -3508,8 +2846,6 @@
     sv_setsv(sv, SWIG_From_int  SWIG_PERL_CALL_ARGS_1((int)(0x06F0)));
     SvREADONLY_on(sv);
   } while(0) /*@SWIG@*/;
-  SWIG_TypeClientData(SWIGTYPE_p_FriBidiRunType, (void*) "Text::Bidi::private::FriBidiRunType");
-  SWIG_TypeClientData(SWIGTYPE_p__FriBidiList, (void*) "Text::Bidi::private::_FriBidiList");
   /*@SWIG:%set_constant@*/ do {
     SV *sv = get_sv((char*) SWIG_prefix "FRIBIDI_MASK_RTL", TRUE | 0x2);
     sv_setsv(sv, SWIG_From_long  SWIG_PERL_CALL_ARGS_1((long)(0x00000001L)));
@@ -3745,11 +3081,6 @@
     sv_setsv(sv, SWIG_From_long  SWIG_PERL_CALL_ARGS_1((long)((0x00000080L+0x00000001L))));
     SvREADONLY_on(sv);
   } while(0) /*@SWIG@*/;
-  /*@SWIG:%set_constant@*/ do {
-    SV *sv = get_sv((char*) SWIG_prefix "FRIBIDI_TYPES_COUNT", TRUE | 0x2);
-    sv_setsv(sv, SWIG_From_int  SWIG_PERL_CALL_ARGS_1((int)(FRIBIDI_TYPES_COUNT)));
-    SvREADONLY_on(sv);
-  } while(0) /*@SWIG@*/;
   ST(0) = &PL_sv_yes;
   XSRETURN(1);
 }
